custom_components/microsoft_todo/sensor.py

```diff
@@ -0,0 +1,158 @@
+import os
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import track_time_interval
+from requests_oauthlib import OAuth2Session
+from requests.adapters import HTTPAdapter
+from urllib3.util import Retry
+from homeassistant.helpers.entity import generate_entity_id
+
+from homeassistant.components.calendar import (
+    PLATFORM_SCHEMA,
+)
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util.json import load_json, save_json
+from homeassistant.util import dt
+from typing import List, Any, Callable
+
+from .outlook_tasks_api import OutlookTasksApi
+
+from .const import (
+    CONF_CLIENT_ID,
+    CONF_CLIENT_SECRET,
+    AUTH_CALLBACK_PATH,
+    TOKEN_URL,
+    SCOPE,
+    MS_TODO_AUTH_FILE,
+    CONF_LIST_NAME,
+    CONF_SCAN_INTERVAL
+)
+
+_LOGGER = logging.getLogger(__name__)
+SCAN_INTERVAL = timedelta(minutes=10)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_CLIENT_ID): cv.string,
+        vol.Required(CONF_CLIENT_SECRET): cv.string,
+        vol.Required(CONF_LIST_NAME): cv.ensure_list,
+        vol.Optional(CONF_SCAN_INTERVAL, default=SCAN_INTERVAL):
+            vol.All(cv.time_period, cv.positive_timedelta),
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None, scan_interval=SCAN_INTERVAL):
```

Comment:
I think it makes sense to stick with the implementation similar to the Todoist integration where the calendar domain is used instead of sensors. Please check the Todoist integration source code https://github.com/home-assistant/core/tree/dev/homeassistant/components/todoist Also, it should be simpler since stuff like API client, etc are already initialized in `calendar.py`. But please let me know if you have a different opinion.

By the way, I wonder if you have seen PR #20. I know the development is going pretty slow (sorry about that) but the feature was partially implemented there. Maybe we can collaborate on this somehow.

Reply:
Yes, I've seen this source code. But in my opinion this is not the right decision,  since 
HA architecture suggests that sensors are defined in "sensors" file, binary_sensors in  "binary_sensors" file and etc. 

If you add my code to the master brunch, I'll add the next update for sensor with "completed task"  attribute, list id  and other ....

thanks for cooperation.

Reply:
> Yes, I've seen this source code. But in my opinion this is not the right decision, since
HA architecture suggests that sensors are defined in "sensors" file, binary_sensors in "binary_sensors" file and etc.

Yes, that's true. But if you check [other to do services integrations](https://www.home-assistant.io/integrations/#calendar) then all of them are in the calendar domain. Todoist, for example, [uses `CalendarEventDevice`](https://github.com/home-assistant/core/blob/dev/homeassistant/components/todoist/calendar.py#L211).